src/registration/forceRegistration.ts

```diff
@@ -35,20 +35,21 @@ export function shouldStatusBarBeProminent(): boolean {
 
 if (isForceEnabled.value === undefined) {
   const disposable = onDidRefreshCapabilities(() => {
-    disposable.dispose();
     isForceEnabled.value = isCapabilityEnabled(Capability.FORCE_REGISTRATION);
+    if (isForceEnabled.value) {
+      disposable.dispose();
```

Comment:
What is this disposable object?

Reply:
this is the design pattern implemnted in vscode